packages/cyberstorm/src/components/Forms/CreateTeamForm/CreateTeamForm.tsx

```diff
@@ -0,0 +1,111 @@
+"use client";
+import { faArrowsRotate } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useController, useForm, Form } from "react-hook-form";
+import * as Button from "../../Button";
+import { TextInput } from "../../TextInput/TextInput";
+import { useToast } from "../../Toast/Provider";
+import styles from "./CreateTeamForm.module.css";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { isRecord } from "../../../utils/type_guards";
+
+interface errorResponse {
+  error: { message: string };
+}
+
+function isErrorResponse(response: unknown): response is errorResponse {
+  return (
+    isRecord(response) &&
+    isRecord(response.error) &&
+    typeof response.error.message === "string"
+  );
+}
+
+export function CreateTeamForm() {
+  const toast = useToast();
+
+  interface formFields {
+    teamName: string;
+  }
+
+  const schema = z.object({
+    teamName: z.string({ required_error: "Team name is required" }),
+  });
+
+  const {
+    control,
+    formState: { isSubmitting },
+  } = useForm<formFields>({
+    mode: "onSubmit",
+    resolver: zodResolver(schema),
+  });
+
+  const teamName = useController({
+    control: control,
+    name: "teamName",
+    rules: {
+      required: "Team name is required",
+    },
+  });
+
+  return (
+    <Form
+      control={control}
+      action="https://thunderstore.io/api/team/create"
+      method="post"
+      onSubmit={() => {
```

Comment:
Seems that when testing locally, the field value is not validated before submit. I.e. entering a character and then removing it (to activate the submit button) lets me submit an empty string to backend. You probably need to check the documentation for some missing piece? Unless the problem is on my local env, I'd consider this a borderline blocker.